components/UI/profileCard/profileCard.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { FunctionComponent, useEffect, useState } from "react";
+import styles from "@styles/dashboard.module.css";
+import CopyIcon from "@components/UI/iconsComponents/icons/copyIcon";
+import ShareIcon from "@components/UI/iconsComponents/icons/shareIcon";
+import { CDNImage } from "@components/cdn/image";
+import { useStarkProfile } from "@starknet-react/core";
+import { minifyAddress, minifyAddressFromStrings } from "@utils/stringService";
+import StarknetIcon from "@components/UI/iconsComponents/icons/starknetIcon";
+import TrophyIcon from "../iconsComponents/icons/trophyIcon";
+import xpUrl from "public/icons/xpBadge.svg";
+import { getCompletedQuests } from "@services/apiService";
+import { decimalToHex } from "@utils/feltService";
+import { getDomainFromAddress } from "@utils/domainService";
+
+
+
+const ProfileCard: FunctionComponent<ProfileCardModified> = ({
+  identity,
+  addressOrDomain,
+  sinceDate,
+  userPercentile,
+  data
+}) => {
+  const [copied, setCopied] = useState(false);
+  const { data: profileData } = useStarkProfile({ address: identity?.addr });
+  const [ displayData, setDisplayData] = useState<FormattedRankingProps>([]);
+
+
+  const copyToClipboard = () => {
```

Comment:
i think this can be added as a util. Can be used globally if we ever need the same functionality